helmfile.d/values/_helpers.tpl

```diff
@@ -90,3 +90,11 @@ after the separator.
 {{- define "gen.container_uri" -}}
 {{ with include "gen.reg-rep-img" . }}{{ . }}{{ end }}{{ if or .tag .digest }}:{{ .tag }}{{ if .digest }}@{{ .digest }}{{ end }}{{ end }}
 {{- end }}
+
+{{- define "componentObjectStorage" -}}
+{{- $component := index . 0 -}}
+{{- $key := index . 1 -}}
+{{- $ctx := index . 2 -}}
+{{- $default := $ctx.Values | dig "objectStorage" "s3" $key "" -}}
+{{- $ctx.Values | dig $component "objectStorage" "s3" $key $default | quote }}
```

Comment:
Are you intending to also make this work for other object storage providers, like azure?